docs/sources/alerting/set-up/provision-alerting-resources/terraform-provisioning/index.md

```diff
@@ -0,0 +1,316 @@
+
+The following is an example you can use to configure the Terraform provider.
+
+```terraform
+terraform {
+    required_providers {
+        grafana = {
+            source = "grafana/grafana"
+            version = ">= 1.28.2"
+        }
+    }
+}
+
+provider "grafana" {
+    url = <YOUR_GRAFANA_URL>
+    auth = <YOUR_GRAFANA_API_KEY>
+}
+```
+
+## Provision contact points and templates
+
+Contact points connect an alerting stack to the outside world. They tell Grafana how to connect to your external systems and where to deliver notifications. There are over fifteen different integrations to choose from.
+
+To provision contact points and templates, complete the following steps.
+
+1. Copy this code block into a .tf file on your local machine.
+
+This example uses a contact point that sends alert notifications to Slack.
+
+```terraform
+resource "grafana_contact_point" "my_slack_contact_point" {
+    name = "Send to My Slack Channel"
+
+    slack {
+        url = <YOUR_SLACK_WEBHOOK_URL>
+        text = <<EOT
+{{ len .Alerts.Firing }} alerts are firing!
+
+Alert summaries:
+{{ range .Alerts.Firing }}
+{{ template "Alert Instance Template" . }}
+{{ end }}
+EOT
+    }
+}
+```
+
+1. Enter text for your notification in the text field.
+
+The `text` field supports [Go-style templating](https://pkg.go.dev/text/template). This enables you to manage your Grafana Alerting message templates directly in Terraform.
+
+1. Run the command ‘terraform apply’.
+
+1. Go to the Grafana UI and check the details of your contact point.
+
+**Note:**
+
+You cannot edit resources provisioned from Terraform from the UI. This ensures that your alerting stack always stays in sync with your code.
+
+1. Click **Test** to verify that the contact point works correctly.
+
+**Note:**
+
+You can re-use the same templates across many contact points. In the example above, a shared template ie embedded using the statement `{{ template “Alert Instance Template” . }}`
+
+This fragment can then be managed separately in Terraform:
+
+```terraform
+resource "grafana_message_template" "my_alert_template" {
+    name = "Alert Instance Template"
+
+    template = <<EOT
+{{ define "Alert Instance Template" }}
+Firing: {{ .Labels.alertname }}
+Silence: {{ .SilenceURL }}
+{{ end }}
+EOT
+}
+```
+
+## Provision notification policies and routing
+
+Notification policies tell Grafana how to route alert instances, as opposed to where. They connect firing alerts to your previously defined contact points using a system of labels and matchers.
+
+To provision notification policies and routing, complete the following steps.
+
+1. Copy this code block into a .tf file on your local machine.
+
+In this example, the alerts are grouped by `alertname`, which means that any notifications coming from alerts which share the same name, are grouped into the same Slack message.
+
```

Comment:
The example also shows nested policies, so it may be worth mentioning here. Perhaps something like:

"If you want to route specific notifications differently, you can add sub-policies. Sub-policies allow you to apply routing to different alerts based on label matching. In this example, we apply a mute timing to all alerts with the label `a=b`